### PR TITLE
build: add Compose-specific detekt rules, dependency analysis, and more strict detekt rules

### PR DESCRIPTION
**Pull request by Vladyslav-Soldatenko**
## Description

Follow-up on the lint/static-analysis hardening investigation: wires in three more layers of checking and fixes everything they found.

## Type of Change

- [ ] New feature
- [ ] Bug fix
- [ ] Refactoring (no functional changes)
- [ ] Documentation update
- [x] Infrastructure / CI/CD
- [ ] Performance improvement

## Changes Made

- Add [mrmans0n/compose-rules](https://github.com/mrmans0n/compose-rules) (`io.nlopez.compose.rules:detekt:0.4.28`, pinned specifically to match this project's detekt 1.23.8 — later versions require detekt 2.0.0-alpha) as a second, complementary Compose-specific detekt ruleset alongside Slack's Android-Lint-based `compose-lint-checks`. Different implementation, different smells (missing `remember`, missing `Modifier` params, multiple content emitters, past-tense callback naming, etc.).
- Add the [Dependency Analysis Gradle Plugin](https://github.com/autonomousapps/dependency-analysis-gradle-plugin) (3.18.0) and wire `projectHealth` into `check`.
- Flip on several more default-off detekt rules across `complexity`, `potential-bugs`, and `style` (see `detekt.yml`), including `ForbiddenSuppress` to stop blanket `@Suppress` usage on the two null-safety rules.
- **Fixed everything the new checks found:**
  - Grouped large decimal literals in tests with underscores (`UnderscoresInNumericLiterals`).
  - Renamed past-tense Compose lambda parameters to present tense (`onXSelected` → `onXSelect`, `onToggleEnabled` → `onEnabledChange`, etc.) per `ParameterNaming` — left the *ViewModel methods* of the same name untouched since the rule only applies to `@Composable` parameters, not regular class methods. Suppressed `WordListContent`'s `onDeleteSelected` specifically with a rationale comment: "Selected" there describes the already-selected items being deleted, not a past-tense event report.
  - Renamed `@Preview` functions so "Preview" is a true suffix (`SettingsScreenPreview_AllGranted` → `SettingsScreenAllGrantedPreview`) per `PreviewNaming`, matching this project's own documented convention.
  - Wrapped `LearningCardFooter`'s multi-emitter branch in a `Column` (`MultipleEmitters`) so it doesn't rely on always being placed inside a `Column` by its caller.
  - Captured `onUnlock`/`onUndoEventShown` via `rememberUpdatedState` before use inside `LaunchedEffect` blocks not keyed on them (`LambdaParameterInRestartableEffect`), avoiding a stale-closure bug if the callback identity ever changes across recomposition.
  - Removed unused dependencies (`androidx.room.ktx`, `androidx.lifecycle.runtime.ktx`, `androidx.test.uiautomator`, and several redundant test-only declarations) and fixed two `runtimeOnly` misconfigurations — verified with a full test run after each removal, not just taken on the tool's word.
  - Excluded `zstd-jni` and `openai-java` from the unused-dependency check with inline rationale, since both are genuine false positives here: DAGP flagged `testImplementation(zstd-jni)` as redundant, but removing it broke `AnkiApkgVocabularyParserTest` with a real failure (the `@aar`-classified main dependency doesn't provide a JVM-loadable native binding); `openai-java` is an umbrella artifact whose used classes live in its `-core`/`-client-okhttp` children, so DAGP's literal "remove it" advice would just break the build.
  - **`LabeledExpression`** was evaluated but deliberately left inactive: all 23 flagged sites were idiomatic `return@launch`/`return@withLock`/`return@withContext`-style early exits in core ViewModels/repositories, not real complexity smells — confirmed with the repo owner before proceeding.

⚠️ **Not included:** a CI step for `projectHealth` (this session's push token lacks `workflow` scope to modify `.github/workflows/ci.yml`). Diff below to add manually:
```diff
       - name: Android Lint
         run: ./gradlew lint --no-daemon
 
+      - name: Dependency analysis
+        run: ./gradlew projectHealth --no-daemon
+
       - name: Unit tests
         run: ./gradlew testDebugUnitTest --no-daemon
@@ (after the lint-reports upload step)
+
+      - name: Upload dependency analysis report
+        if: failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: dependency-analysis-reports
+          path: app/build/reports/dependency-analysis/project-health-report.txt
```

## How to Test

1. `./gradlew check` — runs ktlint, detekt (now including compose-rules), Android Lint, `projectHealth`, and unit tests all together.
2. `./gradlew testDebugUnitTest --tests "*AnkiApkgVocabularyParserTest"` specifically exercises the native zstd JNI path that motivated the `zstd-jni` exclude.

## Checklist

- [x] Code follows the project's style guidelines
- [x] Self-reviewed the code for obvious errors
- [x] Added or updated tests where applicable, including edge cases
- [x] Existing tests pass locally
- [ ] Updated documentation if needed
- [x] No duplication introduced. I searched the existing code for similar behavior and extracted similar parts to be reused
- [x] No new warnings or supressions introduced. If something like this has to be introduced - get approve from maintainer first — the two suppressions added (`ForbiddenSuppress`'s exclude list and `onDeleteSelected`'s `@Suppress`) both carry inline rationale, and the `LabeledExpression` decision was confirmed with the repo owner directly.

## Related Issues

Follow-up to the earlier lint/static-analysis hardening investigation.

---
_Generated by [Claude Code](https://claude.ai/code/session_01URbbDFNKHkSb1k1S8TCda6)_

### CODE REVIEW

File: detekt.yml [hunk 1]
```diff
@@ -22,6 +22,15 @@ complexity:
...
         active: true
+    ComplexInterface:
+        active: true
+        # Room @Dao interfaces legitimately accumulate many small single-purpose query methods;
+        # match the TooManyFunctions.thresholdInClasses tuning above instead of forcing DAO splits.
+        threshold: 20
+    # LabeledExpression intentionally left at detekt's default (inactive): every occurrence in this
+    # codebase was an idiomatic return@launch/return@withLock/return@withContext-style early exit,
+    # not an actual complexity smell, and detekt has no way to scope the check without a blanket
+    # exemption list that would also swallow genuinely confusing labels in future code.
```
Comment (Vladyslav-Soldatenko) [inline detekt.yml]: remove this comment
